core/ipfsport/peer.go

```diff
@@ -65,13 +66,37 @@ func (pm *PeerManager) getPeerPort() uint16 {
 	defer pm.lock.Unlock()
 	for i, status := range pm.ps {
 		if !status {
-			pm.ps[i] = true
-			return pm.startPort + uint16(i)
+			port := pm.startPort + uint16(i)
+			availability := isPortAvailable(port)
+
+			if availability {
+				fmt.Println("available port: ", port, availability)
+				pm.ps[i] = true
+				return pm.startPort + uint16(i)
+			} else {
+				fmt.Println("NOT available port: ", port, availability)
+			}
```

Comment:
Is these print necessary?

Reply:
We can remove the print statements once the port unavailability is tested on MacOs.